src/app/listener.ts

```diff
@@ -1,19 +1,20 @@
-import { IContainer, IMessage, IHandler } from '../common/types';
+import { IContainer, IHandler, IMessage } from '../common/types';
 
 export class Listener {
-  private _handlers: IHandler[] = [];
+  private _messageHandlers: IHandler[] = [];
+  private _channelHandlers: IHandler[] = [];
 
   constructor(public container: IContainer) {
     this._initializeHandlers();
-    // TODO (joey-colon): Abstract this into its own ChannelHandler class where all channel event logic is held.
-    this.container.clientService.on('channelCreate', () => {
-      this.container.classService.updateClasses();
+
+    this.container.clientService.on('channelCreate', async () => {
+      this._executeHandlers(this._channelHandlers);
     });
-    this.container.clientService.on('channelDelete', () => {
-      this.container.classService.updateClasses();
+    this.container.clientService.on('channelDelete', async () => {
+      this._executeHandlers(this._channelHandlers);
     });
-    this.container.clientService.on('channelUpdate', () => {
-      this.container.classService.updateClasses();
+    this.container.clientService.on('channelUpdate', async () => {
+      this._executeHandlers(this._channelHandlers);
     });
```

Comment:
why is this abstraction necessary?

Reply:
in case we want to ever have any sort of other events to react upon channel events. `listener.ts` shouldnt ever be responsible for invoking `updateClasses()` directly imo, it should just be responsible for running the `execute()` method for all instances that are within our `channelHandlers` (type `IHandler`) array (which will include the classes handler)